src/main/java/hudson/plugins/descriptionsetter/DescriptionSetterHelper.java

```diff
@@ -69,9 +72,17 @@ public static boolean setDescription(AbstractBuild<?, ?> build,
 				result = getExpandedDescription(matcher, description);
 				result = build.getEnvironment(listener).expand(result);
 			} else {
-				if (result == null && regexp == null && description != null) {
-					result = description;
-				}
+				if (result == null && (regexp == null || regexp.isEmpty()) && description != null) {
+                    if (Jenkins.getInstance().getPlugin("token-macro") != null) {
+                        try {
+                            result = TokenMacro.expandAll(build, listener, description);
```

Comment:
This will blow up when the plugin is not installed. Given how many plugins require `tokenmacro`, I suggest not making it optional.